gcc/testsuite/rust/compile/stringify_macro.rs

```diff
@@ -0,0 +1,16 @@
+#![feature(rustc_attrs)]
+
+#[rustc_builtin_macro]
+macro_rules! stringify {
+    () => {};
+}
+
+macro_rules! a {
+    () => {
+        " foo"
+    };
+}
+
+fn main() {
+    let _a = stringify!(a!());
```

Comment:
Nor sure if it makes sense but you could do a dg-ouput test case to call printf to verify the string but otherwise this LGTM